docs/content/standards/kiosk-apps/nft-rental.mdx

```diff
@@ -0,0 +1,418 @@
+
+## Smart contract design
+
+:::warning
+
+Transferring kiosks might result in unexpected behaviors while an asset is being rented. If you want to disallow kiosk transferring all together, consider using personal kiosks.
+
+:::
+
+The smart contract design leverages the Kiosk Apps standard to create a framework enabling transactions between lenders and borrowers. Both parties are required to install this extension to facilitate lending or borrowing activities. Moreover, the creator of the rented asset must establish a rental policy and a `ProtectedTP` object, granting authorization to the renting extension for managing assets subject to royalty enforcement.
+
+:::info
+
+This implementation is charging a rental fee based on days. You can re-purpose and update the logic to support charging per hour, or even seconds.
+
+:::
+
+## Move modules
+
+The NFT Rental example uses a single module, `nft_rental.move`. You can find the source for this file hosted in the [`sui-nft-rental` repository](https://github.com/MystenLabs/sui-nft-rental/blob/main/move/nft_rental/sources/nft_rental.move) on GitHub. The source code includes extensive comments to help you follow the example's logic and structure. 
+
+### nft_rental
+
+The `nft_rental` module provides an API that facilitates lending or borrowing through the following operations:
+
+- List for renting
+- Delist from renting
+- Rent
+- Borrow by reference and borrow by value
+- Reclaim for the lender
+
+### Structs
+
+The object model of the `nft_rental` module provides the structure of the app, beginning with the `Rentables` object. The struct only has the `drop` ability and acts as the extension key for the Kiosk `Rentables` extension.
+
+```move
+struct Rentables has drop {}
+```
+
+The `Rented` struct represents a rented item. The only field the struct includes is the ID of the object. 
+
+```move
+struct Rented has store, copy, drop { 
+    id: ID 
+}
+```
+
+The `Listed` struct represents a listed item. The only field the struct includes is the ID of the object.
+
+```move
+struct Listed has store, copy, drop { 
+    id: ID 
+}
+```
+
+The `Promise` struct is created for borrowing by value. The `Promise` operates as the [hot potato](../../concepts/sui-move-concepts/patterns/hot-potato.mdx) that can only be resolved by returning the item back to the extension's `Bag`.
```

Comment:
Hot potatoes typically don't have `store`, because the inclusion of that ability allows someone to avoid the obligation of the hot potato by storing it in some other object, and then forgetting about it.